### DISCUSSION TITLE
How to disable plugin-git while dev mode

### DISCUSSION THREAD

**Discussion by syuilo** [vuepress/core#1038] [Q&A] [answered]
The documentation includes the following tips:
> This plugin will significantly slow down the speed of data preparation, especially when you have a lot of pages. You can consider disabling this plugin in dev mode to get better development experience.

However, there is no specific way to do this.
You can get information on whether VuePress is in dev mode with app.env.isDev, but `app` is not available in `config -> theme -> themePlugins`. Therefore, I can't find a way to disable plugin-git in dev mode.

**Comment by Mister-Hope** [comment, marked as answer]
? Doesn't a theme receive app as params?

**Reply by meteorlxy** [reply]
Well, I think he is talking about the config file. Then the following snippet in the config of our docs would help:

https://github.com/vuepress/vuepress-next/blob/d1b2b63d5ffe0af810ce8aa69604f767f70d300d/docs/.vuepress/config.ts#L20

Also, I think we can support something like this:

```ts
defineUserConfig(({ command }: { command: 'dev' | 'build' }) => ({
   // config..
}))
```

**Reply by Mister-Hope** [reply]
I really don’t think that’s necessary, if you want to fix the problem, support (app)=> Plugin[] shall help

**Reply by Mister-Hope** [reply]
Users do use a Jacky way to access app in config file now
